.github/workflows/test_tap.yml

```diff
@@ -0,0 +1,35 @@
+name: Test tap-github
+
+on: [push, pull_request]
+
+jobs:
+  tests:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
+      with:
+        # Version of Poetry to use
+        version: 1.1.8
+    - name: Install dependencies
+      run: |
+        poetry install
+    - name: Check formatting with black
+      run: |
+        poetry run black --check .
+    - name: Check typing annotations with mypy
+      run: |
+        poetry run mypy . --ignore-missing-imports
+    - name: Test with pytest
+      run: |
+        poetry run pytest
```

Comment:
@laurentS If the first one fails, do the other checks still run?

Reply:
No, it stops immediately. Github actions will also abort all jobs in the matrix if one of them fails, for instance if python 3.6 fails, all parallel jobs will be cancelled no matter how far they got (unless they completed before). You can see an example of this [here](https://github.com/laurentS/slowapi/runs/3450829365?check_suite_focus=true) (I just reused the code I wrote there, the CI definition is exactly the same).
I don't know if there's a way to change that behaviour. Given that the tests run for about a minute, I haven't bothered digging :)

Reply:
> if python 3.6 fails, all parallel jobs will be cancelled no matter how far they got

I didn't know that, thanks for sharing!

> the tests run for about a minute

You're right in that. And the order is OK then. First code style, then type annotations, then tests 👍 

cc @aaronsteers